### ISSUE TITLE
[question] rtree support

### ISSUE THREAD

**Issue by moovida** [simolus3/moor#655]
While trying to do spatial range queries I am getting an exception:
```
Unhandled Exception: SqliteException(1): no such module: rtree
```

More about it: https://www.sqlite.org/rtree.html (section 2)

Is there a possibility to get this enabled?
Spatial queries on plain sqlite would be a big deal.

Actually the strange thing is that the tests on macos pass, so this is currently an android issue (could not check on IOS yet).

Thanks for your comment.

**Comment by moovida** [comment]
One additional question. Is there a wait to make for example an update work when the trigger is in "after" mode?

I.e. I am not able to update a column because the trigger fails due to missing rtree module. Is it possible to do something to have the update end anyways?

Since in this case the rtree is here to "just" speed up things, I would like to allow the user to change values anyways.

**Comment by simolus3** [comment]
> Is there a possibility to get this enabled?

Good idea. I've enabled rtree support on Android with https://github.com/simolus3/sqlite-native-libraries/commit/787cbaa9e190140a1c40e9084455f2fdcc079ae3 and e6a51b5ac4e3572208870c8f944587854e673ac2. 

I'm not aware of a way to make triggers optional. But with that update it shouldn't be a problem anymore.

**Comment by moovida** [comment]
I am touched, thanks a ton. I will let you know how it works. I confirm that on IOS this works.

Just to make sure I understood. Does this mean that I an already test it using the develop branch?

**Comment by simolus3** [comment]
> Does this mean that I an already test it using the develop branch?

Yes, you can just use `moor_ffi` from develop:

```yaml
dependency_overrides:
  moor_ffi:
    git:
      url: https://github.com/simolus3/moor.git
      ref: develop
      path: moor_ffi
```

**Comment by moovida** [comment]
Just wanted to confirm that it works great, thanks again.

One question, I see in the commit "0.7.0-dev". Does that mean it will be included in the 0.7.0 release?

**Comment by simolus3** [comment]
Yes. `moor_ffi` version `0.7.0` will include the rtree module. I'll try to release that update this weekend.

**Comment by simolus3** [comment]
FYI, I've just released `moor_ffi` version `0.7.0`. So you can change the git dependency back to a the regular one if you want to.

**Comment by moovida** [comment]
Thanks a lot, that helps to release the packages to pub.
